daemon/main.go

```diff
@@ -500,6 +501,18 @@ func initEnv(cmd *cobra.Command) {
 	log.Info("|___|_|_|_|___|_|_|_|")
 	log.Infof("Cilium %s", version.Version)
 
+	envoyVersion := envoy.GetEnvoyVersion()
+	log.Infof("%s", envoyVersion)
+
+	envoyVersionArray := strings.Fields(envoyVersion)
+	if len(envoyVersionArray) < 3 {
+		log.Fatal("Truncated Envoy version string, cannot verify version match.")
+	}
+	// Make sure Envoy version matches ours
+	if !strings.HasPrefix(envoyVersionArray[2], version.GetCiliumVersion().Revision) {
+		log.Fatal("Envoy version mismatch, aborting.")
```

Comment:
If we make the mismatch issue just a warning we don't need a disableEnvoyVersionCheck flag thing just for tests (unless we allow both the prod version and whatever the tests provide). It would also meet aanm's requirement to allow a developer to run a modified version. Up to you, though.

Reply:
Yeah, I think this is the best option overall.

Reply:
@tgraf @joestringer On a closer look this does not affect unit testing at all, and bash tests seem be fine in this regard as well. The test failure is due to a false positive potential deadlock due to kv-store transaction(s) taking a long time. The forthcoming kv-store improvements should allow us to remove the kv-store roundtrips we now do while holding a buildmutex.

To verify this I'll ask for re-test. If it passes then we could also keep the version mismatch as a fatal event, at least for now so that we catch any situations where it could happen?